packages/protocol/contracts/layer1/preconf/impl/PreconfRouter2.sol

```diff
@@ -0,0 +1,229 @@
+            urc.isOptedIntoSlasher(_lookaheadSlot.registrationRoot, preconfSlasher),
+            OperatorIsNotOptedIn()
+        );
+    }
+
+    /// @dev Validates if the provided lookahead data points to the current preconfing period.
+    function _validatePreconfingPeriod(
+        uint256 _epochTimestamp,
+        uint256 _currentLookaheadIndex,
+        bytes26 _currentLookaheadHash,
+        ILookaheadStore.LookaheadSlot[] memory _previousLookahead,
+        ILookaheadStore.LookaheadSlot[] memory _currentLookahead,
+        ILookaheadStore.LookaheadSlot memory _lookaheadSlot
+    )
+        internal
+        view
+    {
+        // Validate the current lookahead data
+        require(
+            LibPreconfUtils.calculateLookaheadHash(_epochTimestamp, _currentLookahead)
+                == _currentLookaheadHash,
+            InvalidCurrentLookahead()
+        );
+
+        if (_currentLookaheadIndex != 0) {
+            // This is the case when the preconfer does not have the first preconfing slot in this
+            // epoch.
+            //
+            // Eg: [ x x x x Pa x x x Pb x x x]
+            // - Pb is our preconfer for this epoch.
+            // - Pa is the preconfer at `previousLookaheadSlot`.
+            // - x represents empty slots with no opted in preconfer.
+
+            ILookaheadStore.LookaheadSlot memory previousLookaheadSlot =
+                _currentLookahead[_currentLookaheadIndex - 1];
+
+            // Validate the preconfing period
+            require(
+                block.timestamp > previousLookaheadSlot.slotTimestamp
+                    && block.timestamp <= _lookaheadSlot.slotTimestamp,
+                InvalidLookaheadTimestamp()
+            );
+        } else {
+            // This is the case when the preconfer does have the first preconfing slot in this
+            // epoch.
+            //
+            // Eg: [ prev-epoch ] [ x x x Pa x x x ]
+            // - Pa is our preconfer for this epoch.
+            // - x represents empty slots with no opted in preconfer.
+            //
+            // This opens up three scenarios:
+            // 1. prev-epoch lookahead had 1 or more preconfers
+            // 2. prev-epoch lookahead was empty
+            // 3. prev-epoch lookahead was not posted
+            //
+            // For case 2 and 3, we leave _previousLookahead as an empty array.
+
+            if (_previousLookahead.length != 0) {
+                // Validate previous lookahead data
+                uint256 previousEpochTimestamp =
+                    _epochTimestamp - LibPreconfConstants.SECONDS_IN_EPOCH;
+                bytes26 previousLookaheadHash =
+                    lookaheadStore.getLookaheadHash(previousEpochTimestamp);
+                require(
+                    LibPreconfUtils.calculateLookaheadHash(
+                        previousEpochTimestamp, _previousLookahead
+                    ) == previousLookaheadHash,
+                    InvalidPreviousLookahead()
```

Comment:
Shouldn't we do this check _before_ doing `if (_previousLookahead.length != 0)` in case `_previousLookahead` is wrong?